File: compiler/bench/partition.fs
\ expected: 190569292
\ Integer partition of 100

: partition ( n max -- count ) recursive
  over 0= if 2drop 1 exit then
  over 0< if 2drop 0 exit then
  dup 0= if 2drop 0 exit then
  2dup - over recurse
  swap 1- recurse + ;

: main
  100 100 partition . cr ;

File: compiler/bench/partition.c
// expected: 190569292
#include <stdio.h>
long partition(long n, long max) {
    if (n == 0) return 1;
    if (n < 0) return 0;
    if (max == 0) return 0;
    return partition(n - max, max) + partition(n, max - 1);
}

int main() {
    printf("%ld\n", partition(100, 100));
    return 0;
}
